PP-668/update-devportal (#2)
franciscotobar/devportal: rif/relay/architecture.md

### Smart Wallet
It’s the `contract-based account` owned by the Requester’s EOA. Before executing any transaction using the smart wallet, the smart wallet contract needs to be deployed. 

Smart Wallet are contracts that verify forwarded data and subsequently invoke the recipient contract of the transaction. Creating smart wallets does not have any gas cost providing the advantage that can be deployed only when necessary. 

It is the component that calls the Recipient contract (i.e, the `msg.sender` address the Recipient will see). During the execution, the contract verifies the Relay Request and, if it’s valid, it calls the defined Recipient’s function, otherwise it reverts the invocation. The verification includes checking that the owner of the SmartWallet made the request, rejecting any request with an invalid signature, and preventing replay attacks using a nonce.

The `smart wallet` was designed to only interact with the RIF Relay system, therefore any native currency balance will be transferered back to the owner of `smart wallet` after each transaction. 

### Native Holder Smart Wallet

> The `native holder smart wallet` it's a `smart wallet` that was designed to have interactions outside the RIF Relay system. This means that can hold native currency as it's name describe it.

Reviewer: ```suggestion
The `native holder smart wallet` is a `smart wallet` that was designed to have interactions outside the RIF Relay system. This means that it can hold native currency as it's name describes.
```

Author: updated

> The behavior of the `native holder smart wallet` its the same as the `smart wallet` with the difference that the native currency will not be transferred back to the owner after each transaction and can dispose the usage of the native currency.

Reviewer: ```suggestion
The behavior of the `native holder smart wallet` is the same as the `smart wallet` with the difference that the native currency will not be transferred back to the owner after each transaction and can dispose the usage of the native currency.
```

Author: updated

### Custom Smart Wallet

> The `custom smart wallet` it's a `smart wallet` that was designed to execute custom logic after each transaction. The custom logic it's created at the moment of the contract deployment and can be executed on each transaction. 

Reviewer: ```suggestion
The `custom smart wallet` is a `smart wallet` that was designed to execute custom logic after each transaction. The custom logic is created at the moment of the contract deployment and can be executed on each transaction. 
```

Author: updated

### Tools

### Relay Client

> This is a typescript library to interact with the RIF Relay system. It provides APIs to find a relay, and to send transactions through it. It also expose methods to interact with the blockchain.

Reviewer: ```suggestion
This is a typescript library to interact with the RIF Relay system. It provides APIs to find a relay, and to send transactions through it. It also exposes methods to interact with the blockchain.
```

Author: updated
